lib/vagrant/compose/config.rb

```diff
@@ -22,7 +22,7 @@ def initialize
         @cluster = nil
     		@nodes = {}
         @ansible_groups ={}
-        @multimachine_filter = ((['up', 'provision'].include? ARGV[0]) && ARGV.length > 1) ? ARGV.drop(1) : [] # detect if running vagrant up/provision MACHINE
+        @multimachine_filter = ((['up', 'provision'].include? ARGV[0]) && ARGV.length > 1 && !ARGV[1].start_with?('-')) ? ARGV.drop(1) : [] # detect if running vagrant up/provision MACHINE
```

Comment:
Ok, but I will pick up from here to improve it a little bit (e.g. handle use cases with both options and vm filters, make tolerant to order of options a,d filters etc. etc.)